tools/bgpreader.c

```diff
@@ -256,8 +286,16 @@ int main(int argc, char *argv[])
   /* allocate memory for bs_record */
   bgpstream_record_t *bs_record = NULL;
 
+  /* build the short options */
+  char short_options[LONG_OPTIONS_CNT * sizeof(int)] = {0};
```

Comment:
> Maybe move this declaration to immediately after the declaration of ```LONG_OPTIONS_CNT```.

Fixed. 

> Oh now I see. You're allocating a string to hold the short options. In that case don't you need at most LONG_OPTIONS_CNT * 2 characters? (plus 1 for the \0)

You are totally right. It was a quite 'interesting bug' because if ```LONG_OPTIONS_CNT``` only counts the Non-NULL options without closing parentheses:
```#define LONG_OPTIONS_CNT  (sizeof(LONG_OPTIONS)/sizeof(struct option)) - 1```,
the calculation fails, so that ```LONG_OPTIONS_CNT * 2 + 1```is not enough memory. Fixed.